Qualisys/QTM-Unity-Realtime-Streaming/Helpers/RTStreamThread.cs

```diff
@@ -346,13 +349,79 @@ static bool Get6DOFSettings(RTState state, RTProtocol mProtocol)
                     state.bodies.Add(newbody);
 
                 }
-
                 return true;
             }
-
             return false;
         }
 
+        static bool GetForceSettings(RTState state, RTProtocol mProtocol)
+        {
+            bool getStatus = mProtocol.GetForceSettings();
+
+            if (!getStatus)
+            { 
+                return false;
+            }
+                
+
+            var forcePlates = mProtocol.ForceSettings.Plates;
+            state.forceVectors.Clear(); 
+            foreach (var forcePlate in forcePlates)
+            {
+                var forceVector = new ForceVector();
+                forceVector.Force = Vector3.zero;
+                forceVector.Moment = Vector3.zero;
+                forceVector.ApplicationPoint = Vector3.zero;
+                forceVector.Name = forcePlate.Name;
+                var corners = new Vector3[]{
+                    forcePlate.Location.Corner1.QtmRhsToUnityLhs(state.coordinateSystemChange),
+                    forcePlate.Location.Corner2.QtmRhsToUnityLhs(state.coordinateSystemChange),
+                    forcePlate.Location.Corner3.QtmRhsToUnityLhs(state.coordinateSystemChange),
+                    forcePlate.Location.Corner4.QtmRhsToUnityLhs(state.coordinateSystemChange),
+                };
+
+                var zAxis = (
+                    ((corners[3] + corners[2])-(corners[0] + corners[1])) / 2.0f
+                ).normalized;
+
+                var xAxis = (
+                    ((corners[1] + corners[2]) - (corners[0] + corners[3])) / 2.0f
+                ).normalized;
+
+                var yAxis = Vector3.Cross(zAxis, xAxis).normalized;
+
+                var translation = (corners[0] + corners[1] + corners[2] + corners[3]) / 4.0f;
+                Matrix4x4 m = Matrix4x4.identity;
+                
+                m.m00 = xAxis.x;
+                m.m01 = xAxis.y;
+                m.m02 = xAxis.z;
+
+                m.m10 = yAxis.x;
+                m.m11 = yAxis.y;
+                m.m12 = yAxis.z;
+
+                m.m20 = zAxis.x;
+                m.m21 = zAxis.y;
+                m.m22 = zAxis.z;
+
+                m.m03 = translation.x;
+                m.m13 = translation.y;
+                m.m23 = translation.z;

```

Comment:
I think this code should have a comment stating in what way the coordinate system needs to be changed to make it correct. That way, it gets easier to reason about the code and if it does what it should.

Reply:
I saw now that this was added in a later commit. Resolving.